### PR TITLE
scene group

### PR DESCRIPTION
**Pull request by ameerabuf**

### CODE REVIEW

File: src/controller.js [hunk 1]
```diff
@@ -92,58 +93,68 @@ export function getController (config) {
...
    * Prepare scenes data.
    */
-  _config.scenes = config.scenes.map((scene, index) => {
-    scene.index = index;
-
-    if (scene.viewSource && (typeof scene.duration === 'string' || scene.start?.name)) {
-      scene = getTransformedScene(scene, root, viewportSize, horizontal, absoluteOffsetContext);
-
+  _config.scenes = Object.values(
+    config.scenes.reduce((acc, scene, index) => {
```
Comment (ydaniv) [inline src/controller.js:97]: Let's get wild and try using [`Object.groupBy()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy)?

File: src/view.js [hunk 2]
```diff
@@ -357,17 +361,30 @@ export function getTransformedScene (scene, root, viewportSize, isHorizontal, ab
...
+  const transformedScenes = scenes.map(scene => transformSceneRangesToOffsets(
     scene,
     {start: elementLayoutStart, end: elementLayoutStart + size},
     viewportSize,
     isHorizontal,
     absoluteOffsetContext,
     offsetTree
-  );
+  ));
+  transformedScenes.forEach(scene => {scene.isFixed = isFixed;})
```
Comment (ydaniv) [inline src/view.js:372]: Let's fold this loop into the above one

Comment (ameerabuf) [reply src/view.js]: Done

**Review by ydaniv — APPROVED** [review]

File: src/controller.js (hunk 1 above)
Comment (ameerabuf) [reply src/controller.js:98]: waiting for now, added a TODO for finding a polyfill and using groupBy

**Pull request merged**